[fmt, spdlog] "/utf8 error"build failure

### Operating system

Windows

### Compiler

Clang

### Steps to reproduce the behavior

```Shell
like https://github.com/microsoft/vcpkg/issues/40214 
when I use spdlog v1.14.1 it works, but when I upgrade spdlog to v1.15.0, it still have the "\utf8" error
```

### Failure logs

[build] C:\PROGRA~1\LLVM\bin\CLANG_~1.EXE -DFMT_SHARED -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL -DSPDLOG_SHARED_LIB -IC:/Users/YSY_e/Documents/simona2/build -IC:/Users/YSY_e/Documents/simona2/simona2/loger/include -isystem C:/vcpkg/installed/x64-windows/include -O3 -DNDEBUG -std=c++20 -D_DLL -D_MT -Xclang --dependent-lib=msvcrt -Wall -Wextra -Weffc++ -Werror=uninitialized -Werror=return-type -Wconversion -Wsign-compare -Werror=unused-result -Werror=suggest-override -Wzero-as-null-pointer-constant -Wmissing-declarations -Wold-style-cast -Wnon-virtual-dtor -Werror -finput-charset=UTF-8 /utf-8 -MD -MT simona2/loger/CMakeFiles/Logger.dir/src/logger.cpp.obj -MF simona2\loger\CMakeFiles\Logger.dir\src\logger.cpp.obj.d -o simona2/loger/CMakeFiles/Logger.dir/src/logger.cpp.obj -c C:/Users/YSY_e/Documents/simona2/simona2/loger/src/logger.cpp
[build] clang++: error: no such file or directory: '/utf-8'

### Additional context

_No response_

> Build passed with `vcpkg install spdlog:x64-mingw-dynamic`

It didn't solve the problem, still got the /utf8 error
``` shell
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --build c:/Users/YSY_e/Documents/code/simona2/build --config Debug --target Logger --
[build] [1/2   0% :: 0.004] Re-checking globbed directories...
[build] [1/2  50% :: 0.048] Building CXX object simona2/loger/CMakeFiles/Logger.dir/src/logger.cpp.obj
[build] FAILED: simona2/loger/CMakeFiles/Logger.dir/src/logger.cpp.obj 
[build] C:\PROGRA~1\LLVM\bin\CLANG_~1.EXE -DFMT_SHARED -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL -DSPDLOG_SHARED_LIB -IC:/Users/YSY_e/Documents/code/simona2/build -IC:/Users/YSY_e/Documents/code/simona2/simona2/loger/include -isystem C:/vcpkg/installed/x64-windows/include -O0 -g -Xclang -gcodeview -std=c++20 -D_DEBUG -D_DLL -D_MT -Xclang --dependent-lib=msvcrtd -Wall -Wextra -Weffc++ -Werror=uninitialized -Werror=return-type -Wconversion -Wsign-compare -Werror=unused-result -Werror=suggest-override -Wzero-as-null-pointer-constant -Wmissing-declarations -Wold-style-cast -Wnon-virtual-dtor -Werror -finput-charset=UTF-8 /utf-8 -MD -MT simona2/loger/CMakeFiles/Logger.dir/src/logger.cpp.obj -MF simona2\loger\CMakeFiles\Logger.dir\src\logger.cpp.obj.d -o simona2/loger/CMakeFiles/Logger.dir/src/logger.cpp.obj -c C:/Users/YSY_e/Documents/code/simona2/simona2/loger/src/logger.cpp
[build] clang++: error: no such file or directory: '/utf-8'
[build] ninja: build stopped: subcommand failed.
[proc] The command: "C:\Program Files\CMake\bin\cmake.EXE" --build c:/Users/YSY_e/Documents/code/simona2/build --config Debug --target Logger -- exited with code: 1
[driver] Build completed: 00:00:00.090
[build] Build finished with exit code 1

--- a/ports/spdlog/fix-msvc-utf8.patch
+++ b/ports/spdlog/fix-msvc-utf8.patch
@@ -1,0 +1,15 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index c5bc7b8..2fcdfff 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -274,8 +274,8 @@ if(MSVC)
+     target_compile_options(spdlog PRIVATE "/Zc:__cplusplus")
+     target_compile_options(spdlog_header_only INTERFACE "/Zc:__cplusplus")
+     if(SPDLOG_MSVC_UTF8)
+-	    target_compile_options(spdlog PUBLIC "/utf-8")
+-	    target_compile_options(spdlog_header_only INTERFACE "/utf-8")    
++	    target_compile_options(spdlog PUBLIC $<$<AND:$<COMPILE_LANGUAGE:CXX>,$<CXX_COMPILER_ID:MSVC>>:/utf-8>)
++	    target_compile_options(spdlog_header_only INTERFACE $<$<AND:$<COMPILE_LANGUAGE:CXX>,$<CXX_COMPILER_ID:MSVC>>:/utf-8>)    
+      endif()
+ endif()
+ 

--- a/ports/spdlog/portfile.cmake
+++ b/ports/spdlog/portfile.cmake
@@ -4,6 +4,8 @@ vcpkg_from_github(
     REF "v${VERSION}"
     SHA512 3dd98409f4625ae4d46ef5f59a2fc22a6e151a13dba9d37433363e5d84eab7cca73b379eeb637d8f9b1f0f5a42221c0cc9a2a70414dc2b6af6a162e19fba0647
     HEAD_REF v1.x
+    PATCHES
+        fix-msvc-utf8.patch
 )
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS

--- a/ports/spdlog/vcpkg.json
+++ b/ports/spdlog/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "spdlog",
   "version-semver": "1.15.0",
+  "port-version": 1,
   "description": "Very fast, header-only/compiled, C++ logging library.",
   "homepage": "https://github.com/gabime/spdlog",
   "license": "MIT",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -8578,7 +8578,7 @@
     },
     "spdlog": {
       "baseline": "1.15.0",
-      "port-version": 0
+      "port-version": 1
     },
     "spectra": {
       "baseline": "1.0.1",

--- a/versions/s-/spdlog.json
+++ b/versions/s-/spdlog.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "72b3738962d622e47073cf4a0dc0e6ddd29d1544",
+      "version-semver": "1.15.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "6595ef6c86cf6618cede07b4e1bc8b4d6b098b45",
       "version-semver": "1.15.0",
       "port-version": 0

